File: backend/app/schemas/market.py
from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel


class PriceResponse(BaseModel):
    symbol: str
    price: float


class MarketCoinResponse(BaseModel):
    symbol: str
    name: str
    price: float
    market_cap: Optional[float] = None
    volume_24h: Optional[float] = None
    change_24h: Optional[float] = None
    image: Optional[str] = None


class OhlcvEntry(BaseModel):
    time: int
    open: float
    high: float
    low: float
    close: float
    volume: float


class MarketOverviewResponse(BaseModel):
    coins: List[MarketCoinResponse]


class CoinSearchResult(BaseModel):
    symbol: str
    name: str
    market_cap_rank: Optional[int] = None
    thumb: Optional[str] = None

File: frontend/src/lib/types/market.ts
export interface PriceResponse {
  symbol: string;
  price: number;
}

export interface MarketCoinResponse {
  symbol: string;
  name: string;
  price: number;
  market_cap: number | null;
  volume_24h: number | null;
  change_24h: number | null;
  image: string | null;
}

export interface MarketOverviewResponse {
  coins: MarketCoinResponse[];
}

export interface CoinSearchResult {
  symbol: string;
  name: string;
  market_cap_rank: number | null;
  thumb: string | null;
}

export interface OhlcvEntry {
  time: number;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
}
